frontend/public/service-worker.js

```diff
@@ -0,0 +1,38 @@
+const filesToCache = ['/']
+
+const staticCacheName = 'pages-cache-v1'
+
+// Install the files we've identified
+self.addEventListener('install', event => {
```

Comment:
self? Well, how about that, service workers get their own special [global object](https://developer.mozilla.org/en-US/docs/Web/API/Web_Workers_API/Functions_and_classes_available_to_workers). Interesting!